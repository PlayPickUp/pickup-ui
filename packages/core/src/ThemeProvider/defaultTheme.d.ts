import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
