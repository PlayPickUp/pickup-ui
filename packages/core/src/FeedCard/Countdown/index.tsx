import React, { useEffect, useState } from "react";
import moment from "moment";
import { createUseStyles, useTheme } from "react-jss";
import { Countdown as Timer, Pick } from "@playpickup/icons";
import { DefaultTheme, CountdownProps } from "../../types";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  timer: {
    minWidth: 120,
    fontSize: "11px",
    fontColor: theme.colors.grey.dark,
    lineHeight: "18px",
    verticalAlign: "middle",
    float: "left",
  },
  pick: {
    fontSize: "11px",
    fontColor: theme.colors.purple.base,
    lineHeight: "18px",
    verticalAlign: "middle",
    float: "left",
  },
  container: {
    float: "left",
    width: 223,
    [theme.mediaQuery(theme.breakpoints.medium)]: {
      width: 300,
    },
  },
  timerIcon: {
    verticalAlign: "middle",
    color: theme.colors.grey.dark,
    paddingRight: "3px",
    height: theme.spacing.base * 5,
    width: theme.spacing.base * 5,
    [theme.mediaQuery(theme.breakpoints.medium)]: {
      height: theme.spacing.base * 6,
      width: theme.spacing.base * 6,
    },
  },
  pickedIcon: {
    verticalAlign: "middle",
    color: (picks: CountdownProps) =>
      picks.has_fan_pick ? theme.colors.purple.dark : theme.colors.grey.dark,
    height: theme.spacing.base * 5,
    width: theme.spacing.base * 5,
    [theme.mediaQuery(theme.breakpoints.medium)]: {
      height: theme.spacing.base * 6,
      width: theme.spacing.base * 6,
    },
  },
  closed: {
    color: "red",
  },
  picked: {
    color: (picks: CountdownProps) =>
      picks.has_fan_pick ? theme.colors.purple.dark : theme.colors.grey.dark,
  },
}));

const Countdown: React.FC<CountdownProps> = (props) => {
  let timerID: NodeJS.Timeout | undefined;
  const [days, setDays] = useState(0);
  const [hours, setHours] = useState(0);
  const [minutes, setMinutes] = useState(0);
  const [seconds, setSeconds] = useState(0);
  const theme = useTheme<DefaultTheme>();
  const picks: CountdownProps = {
    has_fan_pick: props.has_fan_pick,
    close_at: props.close_at,
  };
  const classes = useStyles({ ...picks, theme });

  useEffect(() => {
    let duration = undefined;
    let days = 0;
    let hours = 0;
    let minutes = 0;
    let seconds = 0;
    timerID = setInterval(() => {
      const end = moment(props.close_at).format("YYYY-MM-DD HH:mm:ss");
      const then: moment.Moment = moment(end, "YYYY-MM-DD HH:mm:ss");
      const now: moment.Moment = moment();
      if (then > now) {
        duration = moment.duration(then.diff(now));
        days = Math.floor(duration.asDays());
        hours = duration.hours();
        minutes = duration.minutes();
        seconds = duration.seconds();
      }

      setDays(days);
      setHours(hours);
      setMinutes(minutes);
      setSeconds(seconds);
    }, 1000);
    return () => {
      clearInterval(timerID);
    };
  }, []);

  const generateTimer = () => {
    return (
      `${days > 0 ? days + "D " : ""}` +
      `${hours > 0 ? hours + "h " : ""}` +
      `${minutes > 0 ? minutes + "m " : ""}` +
      `${seconds > 0 ? seconds + "s " : ""}`
    );
  };

  return (
    <div className={classes.container}>
      <div className={classes.timer}>
        <Timer color="currentColor" className={classes.timerIcon} />
        {hours || minutes || seconds > 0 ? (
          generateTimer()
        ) : (
          <span className={classes.closed}>CLOSED</span>
        )}
      </div>
      <div className={classes.pick}>
        <Pick color="currentColor" className={classes.pickedIcon} />
        <span className={classes.picked}>
          {props.has_fan_pick ? "Picked" : "Not Picked"}
        </span>
      </div>
    </div>
  );
};

export default Countdown;
