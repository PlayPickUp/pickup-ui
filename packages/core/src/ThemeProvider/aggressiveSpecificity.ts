/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
