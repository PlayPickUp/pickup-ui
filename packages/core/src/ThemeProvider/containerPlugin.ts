/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
