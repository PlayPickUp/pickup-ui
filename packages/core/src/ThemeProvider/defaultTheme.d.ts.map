{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}