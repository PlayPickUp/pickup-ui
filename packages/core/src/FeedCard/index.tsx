import React from "react";
import Typography from "../Typography";
import { defaultTheme } from "../ThemeProvider/defaultTheme";
import { createUseStyles } from "react-jss";
import { FeedCardProps, DefaultTheme } from "../types";
import Countdown from "./Countdown";
import Button from "../Button";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  container: {
    display: "block",
    position: "relative",
    width: "100%",
    backgroundColor: "transparent",
    [theme.mediaQuery(theme.breakpoints.medium)]: {
      backgroundColor: theme.colors.white,
      border: `0.5px solid ${theme.colors.grey.light}`,
      borderRadius: theme.borderRadius * 2,
      boxShadow: "0px 5px 20px rgba(0, 0, 0, 0.04)",
      padding: "30px 20px",
    },
  },
  postContent: {
    flex: "1 1 405px",
    padding: "0 20px 20px",
  },
  featuredImage: {
    order: -2,
    "& > div": {
      width: "100%",
      borderRadius: theme.borderRadius,
      backgroundColor: theme.colors.grey.light,
      backgroundSize: "cover",
      backgroundPosition: "center center",
      borderBottomLeftRadius: 0,
      borderBottomRightRadius: 0,
      height: "120px",
      objectFit: "contain",
    },
  },
  expandedFeaturedIgmage: {
    order: -2,
    "& > div": {
      height: "140px",
      borderRadius: theme.borderRadius,
      backgroundColor: theme.colors.grey.light,
      backgroundSize: "cover",
      backgroundPosition: "center center",
      borderBottomLeftRadius: 0,
      borderTopRightRadius: 0,
      width: "140px",
      objectFit: "contain",
    },
  },
  publisherRow: {
    display: "flex",
    position: "relative",
    flexFlow: "row wrap",
    alignItems: "center",
    width: "100%",
    margin:theme.spacing.base
  },
  publisherHeadline: {
    flex: "0 1 calc(100% - 58px)",
  },
  cardContainer: {
    marginTop: theme.spacing.base * 5,
    padding: "0px 20px",
    [theme.mediaQuery(theme.breakpoints.small)]: {
      padding: 0,
    },
  },
  card: {
    backgroundColor: theme.colors.white,
    margin: "10px",
    border: "1px solid #E5E3E8",
    boxShadow: "0px 5px 20px rgba(0, 0, 0, 0.08)",
    borderRadius: "8px",
    width: "220px",
    height: "100%",
    minHeight: "290px",
    display: "flex",
    flexDirection: "column",
  },
  expandedCard: {
    backgroundColor: theme.colors.white,
    margin: "10px",
    border: "1px solid #E5E3E8",
    boxShadow: "0px 5px 20px rgba(0, 0, 0, 0.08)",
    borderRadius: "8px",
    maxWidth: "420px",
    minHeight: "140px",
    height: "100%",
    display: "flex",
    flexDirection: "row",
  },
  button: {
    fontSize: 11,
    minHeight: 24,
    padding: "0 10px",
    display: "inline-block",
    marginRight: theme.spacing.base * 5,
  },
  h3: {
    fontWeight: "normal",
    fontSize: 18,
    letterSpacing: "-0.1px",
    padding: "0 10px",
  },
  publisherImage: {
    width: "16px",
    height: "16px",
    marginLeft: theme.spacing.base * 2,

  },
  body: {
    display: "inline-block",
    margin: theme.spacing.base,
    fontSize: 12,
    marginRight: theme.spacing.base,
  },
}));

const renderFeaturedImage = (featured_img: string) => {
  if (!featured_img || featured_img !== "false") {
    return { backgroundImage: `url(${featured_img})` };
  }
};

const FeedCard: React.FC<FeedCardProps> = ({
  image,
  publisherIcon,
  publisherName,
  publishedAt,
  title,
  timeLeft,
  pickCount,
  picks,
  expanded,
}) => {
  const classes = useStyles();

  return (
    <div className={expanded ? classes.expandedCard : classes.card}>
      <div>
        <a
          className={
            expanded ? classes.expandedFeaturedIgmage : classes.featuredImage
          }
          title={title}
        >
          <div style={renderFeaturedImage(image)} />
        </a>
      </div>

      <div
        style={{
          maxWidth: `${expanded ? "280px" : "200px"}`,
          wordWrap: "break-word",
        }}
      >
        <div className={classes.publisherRow}>
          <img
            src={publisherIcon}
            alt={publisherName}
            className={classes.publisherImage}
          />
          <a
            style={{
              display: "inline-block",
            }}
            title={publisherName}
          >
            <Typography
              variant="body"
              className={classes.body}
              color={"#615E66"}
              useUnescape
            >
              {publisherName}
            </Typography>
          </a>
          <Typography
            variant="body"
            element="span"
            color={defaultTheme.colors.grey.base}
            className={classes.body}
          >
            {publishedAt}
          </Typography>
        </div>

        <a title={title} style={{ color: "inherit" }}>
          <Typography
            variant="heading3"
            element="h3"
            className={classes.h3}
            useUnescape
          >
            {title}
          </Typography>
        </a>

        <div
          style={{
            padding: "8px",
          }}
        >
          {picks.map((pick) => {
            return (
              <Button key={pick.value} color="light" className={classes.button}>
                {pick.label}
              </Button>
            );
          })}
        </div>

        <div
          style={{
            padding: "2px 10px 10px 10px",
          }}
        >
          <Countdown close_at={timeLeft} has_fan_pick={pickCount} />
        </div>
      </div>
    </div>
  );
};
export default FeedCard;
