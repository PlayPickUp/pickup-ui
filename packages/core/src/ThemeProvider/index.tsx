import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
