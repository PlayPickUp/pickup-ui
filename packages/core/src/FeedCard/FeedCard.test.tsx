import React from "react";
import { render, waitFor } from "@testing-library/react";
import jssSerializer from 'jss-snapshot-serializer';
expect.addSnapshotSerializer(jssSerializer);
import ThemeProvider from "../ThemeProvider";
import FeedCard from "./index";

test("Renders and matches snapshot", () => {
  const { container } = render(
    <ThemeProvider>
      <FeedCard
        publisherIcon="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/crossnet-play.jpg"
        publisherName="Prime Time Sports"
        publishedAt="23m"
        title="Chiefs vs. Ravens Who wins?"
        image="https://static.clubs.nfl.com/image/private/t_editorial_landscape_12_desktop/ravens/qhsyhehrwlw6he1nfmq0"
        pickCount={2}
        timeLeft={new Date()}
        picks={[
          { label: "Ravens", value: "Ravens" },
          { label: "Chiefs", value: "Chiefs" },
        ]}
        expanded={false}
      />
      <FeedCard
        publisherIcon="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/crossnet-play.jpg"
        publisherName="Prime Time Sports"
        publishedAt="23m"
        title="Chiefs vs. Ravens Who wins?"
        image="https://static.clubs.nfl.com/image/private/t_editorial_landscape_12_desktop/ravens/qhsyhehrwlw6he1nfmq0"
        pickCount={2}
        timeLeft={new Date()}
        picks={[
          { label: "Ravens", value: "Ravens" },
          { label: "Chiefs", value: "Chiefs" },
        ]}
        expanded={true}
      />
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test("Passes and renders props", () => {
  const { getByTestId, getByRole, getByText } = render(
    <ThemeProvider>
      <FeedCard
        publisherIcon="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/crossnet-play.jpg"
        publisherName="Prime Time Sports"
        publishedAt="23m"
        title="Chiefs vs. Ravens Who wins?"
        image="https://static.clubs.nfl.com/image/private/t_editorial_landscape_12_desktop/ravens/qhsyhehrwlw6he1nfmq0"
        pickCount={2}
        timeLeft={new Date()}
        picks={[
          { label: "Ravens", value: "Ravens" },
          { label: "Chiefs", value: "Chiefs" },
        ]}
        expanded={false}
      />
    </ThemeProvider>
  );
  // image
  expect(getByTestId("featuredImage").getAttribute("style")).toContain(
    "background-image: url(https://static.clubs.nfl.com/image/private/t_editorial_landscape_12_desktop/ravens/qhsyhehrwlw6he1nfmq0)"
  );
  // publisherIcon
  expect(getByRole("img").getAttribute("src")).toEqual(
    "https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/crossnet-play.jpg"
  );
  // publishedAt
  expect(getByText("23m")).toBeTruthy();
  // title
  expect(getByText("Chiefs vs. Ravens Who wins?")).toBeTruthy();
  // pickCount
  expect(getByText("Picked")).toBeTruthy();
  // timeLeft
  expect(getByText("CLOSED")).toBeTruthy();
  // picks
  expect(getByRole("button", { name: /Chiefs/i })).toBeTruthy();
});

test("Shows remaining time when time is left", async () => {
  const tomorrow = new Date();
  tomorrow.setDate(tomorrow.getDate() + 1);
  const { getByText } = render(
    <ThemeProvider>
      <FeedCard
        publisherIcon="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/crossnet-play.jpg"
        publisherName="Prime Time Sports"
        publishedAt="23m"
        title="Chiefs vs. Ravens Who wins?"
        image="https://static.clubs.nfl.com/image/private/t_editorial_landscape_12_desktop/ravens/qhsyhehrwlw6he1nfmq0"
        pickCount={2}
        timeLeft={tomorrow}
        picks={[
          { label: "Ravens", value: "Ravens" },
          { label: "Chiefs", value: "Chiefs" },
        ]}
        expanded={false}
      />
    </ThemeProvider>
  );
  await waitFor(() => expect(getByText("23h 59m")).toBeTruthy());
});
