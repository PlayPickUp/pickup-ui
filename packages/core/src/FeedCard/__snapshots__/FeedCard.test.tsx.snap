// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders and matches snapshot 1`] = `
<div>
  <div
    id="PickUpUI"
  >
    <div
      class="PU--card-0-2-8"
    >
      <div>
        <a
          class="PU--featuredImage-0-2-3"
          title="Chiefs vs. Ravens Who wins?"
        >
          <div
            style="background-image: url(https://static.clubs.nfl.com/image/private/t_editorial_landscape_12_desktop/ravens/qhsyhehrwlw6he1nfmq0);"
          />
        </a>
      </div>
      <div
        class="PU--rightContainer-0-2-14 PU--rightContainer-d7-0-2-19"
      >
        <div
          class="PU--publisherRow-0-2-5 PU--publisherRow-d4-0-2-17"
        >
          <img
            alt="Prime Time Sports"
            class="PU--publisherImage-0-2-12"
            src="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/crossnet-play.jpg"
          />
          <a
            style="display: inline-block;"
            title="Prime Time Sports"
          >
            <p
              class="PU--root-0-2-21 PU--root-d6-0-2-30 PU--body-0-2-28 PU--body-0-2-13"
            >
              Prime Time Sports
            </p>
          </a>
          <span
            class="PU--root-0-2-21 PU--root-d7-0-2-31 PU--body-0-2-28 PU--body-0-2-13"
          >
            23m
          </span>
        </div>
        <a
          title="Chiefs vs. Ravens Who wins?"
        >
          <h3
            class="PU--root-0-2-21 PU--root-d8-0-2-32 PU--heading3-0-2-24 PU--h3-0-2-11 PU--h3-d5-0-2-18"
          >
            Chiefs vs. Ravens Who wins?
          </h3>
        </a>
        <div
          class="PU--pick-0-2-15 PU--pick-d9-0-2-20"
        >
          <button
            class="PU--root-0-2-33 PU--root-d0-0-2-37 PU--light-0-2-35 PU--button-0-2-10"
            data-testid="pickup-button"
          >
            Ravens
          </button>
          <button
            class="PU--root-0-2-33 PU--root-d1-0-2-38 PU--light-0-2-35 PU--button-0-2-10"
            data-testid="pickup-button"
          >
            Chiefs
          </button>
        </div>
        <div
          class="PU--countdown-0-2-16"
        >
          <div
            class="PU--container-0-2-41"
          >
            <div
              class="PU--timer-0-2-39"
            >
              <svg
                class="PU--timerIcon-0-2-42"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22 5.71999L17.4 1.85999L16.11 3.38999L20.71 7.24999L22 5.71999ZM7.88 3.38999L6.6 1.85999L2 5.70999L3.29 7.23998L7.88 3.38999ZM12.5 7.99999H11V14L15.75 16.85L16.5 15.62L12.5 13.25V7.99999ZM12 3.99999C7.03 3.99999 3 8.02999 3 13C3 17.97 7.02 22 12 22C16.97 22 21 17.97 21 13C21 8.02999 16.97 3.99999 12 3.99999ZM12 20C8.13 20 5 16.87 5 13C5 9.12999 8.13 5.99999 12 5.99999C15.87 5.99999 19 9.12999 19 13C19 16.87 15.87 20 12 20Z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="PU--root-0-2-21 PU--root-d9-0-2-48 PU--body-0-2-28 PU--closed-0-2-44"
              >
                CLOSED
              </span>
            </div>
            <div
              class="PU--pick-0-2-40"
            >
              <svg
                class="PU--pickedIcon-0-2-43 PU--pickedIcon-d2-0-2-46"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M13.5118 3.75024L13.0534 10.5167H19.0001L10.4888 21.2498L10.9472 14.4833H5.00049L13.5118 3.75024Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="PU--root-0-2-21 PU--root-d10-0-2-49 PU--body-0-2-28 PU--picked-0-2-45 PU--picked-d3-0-2-47"
              >
                Picked
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="PU--expandedCard-0-2-9"
    >
      <div>
        <a
          class="PU--expandedFeaturedIgmage-0-2-4"
          title="Chiefs vs. Ravens Who wins?"
        >
          <div
            style="background-image: url(https://static.clubs.nfl.com/image/private/t_editorial_landscape_12_desktop/ravens/qhsyhehrwlw6he1nfmq0);"
          />
        </a>
      </div>
      <div
        class="PU--rightContainer-0-2-14 PU--rightContainer-d14-0-2-52"
      >
        <div
          class="PU--publisherRow-0-2-5 PU--publisherRow-d11-0-2-50"
        >
          <img
            alt="Prime Time Sports"
            class="PU--publisherImage-0-2-12"
            src="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/crossnet-play.jpg"
          />
          <a
            style="display: inline-block;"
            title="Prime Time Sports"
          >
            <p
              class="PU--root-0-2-21 PU--root-d11-0-2-54 PU--body-0-2-28 PU--body-0-2-13"
            >
              Prime Time Sports
            </p>
          </a>
          <span
            class="PU--root-0-2-21 PU--root-d12-0-2-55 PU--body-0-2-28 PU--body-0-2-13"
          >
            23m
          </span>
        </div>
        <a
          title="Chiefs vs. Ravens Who wins?"
        >
          <h3
            class="PU--root-0-2-21 PU--root-d13-0-2-56 PU--heading3-0-2-24 PU--h3-0-2-11 PU--h3-d12-0-2-51"
          >
            Chiefs vs. Ravens Who wins?
          </h3>
        </a>
        <div
          class="PU--pick-0-2-15 PU--pick-d16-0-2-53"
        >
          <button
            class="PU--root-0-2-33 PU--root-d2-0-2-57 PU--light-0-2-35 PU--button-0-2-10"
            data-testid="pickup-button"
          >
            Ravens
          </button>
          <button
            class="PU--root-0-2-33 PU--root-d3-0-2-58 PU--light-0-2-35 PU--button-0-2-10"
            data-testid="pickup-button"
          >
            Chiefs
          </button>
        </div>
        <div
          class="PU--countdown-0-2-16"
        >
          <div
            class="PU--container-0-2-41"
          >
            <div
              class="PU--timer-0-2-39"
            >
              <svg
                class="PU--timerIcon-0-2-42"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22 5.71999L17.4 1.85999L16.11 3.38999L20.71 7.24999L22 5.71999ZM7.88 3.38999L6.6 1.85999L2 5.70999L3.29 7.23998L7.88 3.38999ZM12.5 7.99999H11V14L15.75 16.85L16.5 15.62L12.5 13.25V7.99999ZM12 3.99999C7.03 3.99999 3 8.02999 3 13C3 17.97 7.02 22 12 22C16.97 22 21 17.97 21 13C21 8.02999 16.97 3.99999 12 3.99999ZM12 20C8.13 20 5 16.87 5 13C5 9.12999 8.13 5.99999 12 5.99999C15.87 5.99999 19 9.12999 19 13C19 16.87 15.87 20 12 20Z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="PU--root-0-2-21 PU--root-d14-0-2-61 PU--body-0-2-28 PU--closed-0-2-44"
              >
                CLOSED
              </span>
            </div>
            <div
              class="PU--pick-0-2-40"
            >
              <svg
                class="PU--pickedIcon-0-2-43 PU--pickedIcon-d4-0-2-59"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M13.5118 3.75024L13.0534 10.5167H19.0001L10.4888 21.2498L10.9472 14.4833H5.00049L13.5118 3.75024Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="PU--root-0-2-21 PU--root-d15-0-2-62 PU--body-0-2-28 PU--picked-0-2-45 PU--picked-d5-0-2-60"
              >
                Picked
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
